<!--
Copyright (c) 2018 Pilz GmbH & Co. KG

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
-->

<launch>

  <!-- PNoz-Multi settings -->
  <arg name="modbus_server_ip" default="169.254.60.1" />
  <arg name="modbus_server_port" default="502" />
  <arg name="index_of_first_register_to_read" default="512" />
  <arg name="num_registers_to_read" default="4" />

  <node required="true" pkg="prbt_hardware_support" type="pilz_modbus_read_client_node" name="pilz_modbus_read_client_node" output="screen">
    <param name="modbus_server_ip" value="$(arg modbus_server_ip)"/>
    <param name="modbus_server_port" value="$(arg modbus_server_port)"/>
    <param name="index_of_first_register_to_read" value="$(arg index_of_first_register_to_read)"/>
    <param name="num_registers_to_read" value="$(arg num_registers_to_read)"/>
  </node>

</launch>
